modules/nic/variables.tf

```
@@ -0,0 +1,47 @@
+variable "name" {
+  description = "The name of this network interface card."
+  type        = string
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group to create the resources in."
+  type        = string
+}
+
+variable "location" {
+  description = "The location to create the resources in."
+  type        = string
+}
+
+variable "ip_configuration" {
+  description = "A map of IP configuration with supported blockss"
+
+  type = map(object({
+    name                          = string
+    subnet_id                     = optional(string)
+    private_ip_address_version    = optional(string)
+    private_ip_address_allocation = string
+    private_ip_address            = optional(string)
+    primary                       = optional(bool, true)
```

Don't think a default here is necessary when the default is configured below.. Might be up to preferences.

```suggestion
  type = map(object({
    name                          = string
    subnet_id                     = optional(string)
    private_ip_address_version    = optional(string)
    private_ip_address_allocation = string
    private_ip_address            = optional(string)
    primary                       = optional(bool)
```

---

Ah yes i thought it was required but if it's not then there is no need for the default there aswell.